onprc_ehr/resources/views/groupProcessing.html

```diff
@@ -241,6 +319,16 @@
                 groupName = Ext4.isArray(groupName) ? groupName: [groupName];
                 var room = btn.up('panel').down('#room').getValue() || [];
                 room = Ext4.isArray(room) ? room: [room];
+
+                var field = btn.up('panel').down('#animalField');
+                var animalid = field.getValue() || [];
+                if (!Ext4.isEmpty(animalid)){
+                animalid = Ext4.String.trim(animalid);
+                animalid = animalid.replace(/[\s,;]+/g, ';');
+                animalid = animalid.replace(/(^;|;$)/g, '');
+                }
+                animalid = Ext4.isArray(animalid) ? animalid: [animalid];
+
```

Comment:
lines 323-331 duplicated a few times above (lines 94-101, 141-148, etc.) - maybe extract it into a new method.